app/models/concerns/alegre_v2.rb

```diff
@@ -148,28 +163,32 @@ def get_type(project_media)
       type
     end
 
+    def content_hash_for_value(value)
+      value.nil? ? nil : Digest::MD5.hexdigest(value)
+    end
+
     def content_hash(project_media, field)
       if Bot::Alegre::ALL_TEXT_SIMILARITY_FIELDS.include?(field)
-        Digest::MD5.hexdigest(project_media.send(field))
+        content_hash_for_value(project_media.send(field))
+      elsif project_media.is_link?
+        return content_hash_for_value(project_media.media.url)
+      elsif project_media.is_a?(TemporaryProjectMedia)
+        return Rails.cache.read("url_sha:#{project_media.url}")
+      elsif project_media.is_uploaded_media?
+        return project_media.media.file.filename.split(".").first
```

Comment:
@DGaffney how strict the content hash should be? Do you expect no clashes? Just noting here that two upload files can have the same name but completely different contents.

Reply:
I think your example would be ok - if a file has a different context, but an existing content hash, I think that's... actually the ideal case? we'd side-step fingerprinting based on the fact we'd seen that media before and just add context to that case? Maybe something to pin as an explicit test for when we roll this out on QA?

Reply:
Sorry, content, not context :D

Reply:
As per conversation with @computermacgyver and @DGaffney , we confirmed filenames are hashes of the content: https://github.com/meedan/check-api/blob/develop/app/models/media.rb#L35-L41. @DGaffney will just do a last confirmation: Upload two files with the same name but different content and be sure that they have different content hashes.

Reply:
@caiosba confirmed and noted as such on the PR!

Reply:
Thanks!